### DISCUSSION TITLE
Trying to generate a subtitle file

### DISCUSSION THREAD

**Discussion by archz2** [openai/whisper#957] [Q&A]
I am using whisper in google collab to generate subtitle file of an mp3 file. I am getting this error. What should I do?

![image](https://user-images.githubusercontent.com/66735186/218265929-23d98dcf-cc10-4c78-ae92-50c2f40ac283.png)

**Comment by EtienneAb3d** [comment]
Try this code:
https://github.com/EtienneAb3d/WhisperTimeSync/blob/main/transcribe.py

**Comment by archz2** [comment]
Thanks. And if the audio is in Korean and I want to generate english subtitle file?

**Reply by EtienneAb3d** [reply]
Try:

`result = model.transcribe(mp3,language="en")
`

**Comment by archz2** [comment]
Okay actually I used a simpler code.

```
!whisper finaOctaneMaterials.mp3 --language=ko

result = model.transcribe(mp3,language="en") 
```


 It is working now. 

![image](https://user-images.githubusercontent.com/66735186/218268747-6ca2f2ce-b25e-4db4-9e89-f3806e2d2eda.png)


I have three questions. 
1. How do I make it use GPU so that it runs faster?
2. It is directly generating subtitles in Korean instead of English even though I used 'result = model.transcribe(mp3,language="en") '. What should I do so that it directly generates them in English?
3. It is using the 'large' model for Korean also, right?

**Reply by EtienneAb3d** [reply]
1. change your notebook configuration to get a GPU
2. I don't understand what your code is doing. Why do you use `--language=ko`? You do not need to specify the input language. Only the output language.
3. I don't understand what your code is doing. It's up to you to specify it when creating the model. See my own code.

**Reply by archz2** [reply]
Okay what lines do I add to make it use the GPU?

P.s. Got it. 
![image](https://user-images.githubusercontent.com/66735186/218270019-09d2ad1f-20cc-408a-9327-38358f9a09f8.png)


I used the following code now. 

```

import sys
from pathlib import Path
import whisper
from whisper.utils import WriteSRT

modelSize = "large"

if __name__ == "__main__":
    mp3 = sys.argv[1] #"data/finaOctaneMaterials.mp3"
    
    if len(sys.argv) >= 2 :
        modelSize = sys.argv[2]
    
    print("Loading Whisper model: "+modelSize+" ...")
    model = whisper.load_model(modelSize)
    
    print("Transcribing: "+mp3+" ...")
    result = model.transcribe(mp3,language="en") 
    
    print(result["text"])
    
    print("Saving: "+mp3+".srt ...")
    p = Path(mp3)
    writer = WriteSRT(p.parent)
    writer(result, mp3)
```


I got this error. 
![image](https://user-images.githubusercontent.com/66735186/218269926-c5c65de1-bbae-4195-8225-cb6eb3fea406.png)

**Reply by EtienneAb3d** [reply]
Your notebook config looks like mine.

It should display:

`Loading Whisper model: large`

If you are executing the code directly, you should perhaps remove the "main" line.

**Reply by archz2** [reply]
I don't understand. What should be the new lines of code now? I ran your code by copy pasting in Google Collab.

**Reply by EtienneAb3d** [reply]
With the "main" line, this code is written to be used as a command line, like:

`python3 transcribe.py data/KatyPerry-Firework.mp3 large`

To use it directly in interactive mode, you should remove the "main" line, like:
```
import sys
from pathlib import Path
import whisper
from whisper.utils import WriteSRT

modelSize = "large"

mp3 = "data/KatyPerry-Firework.mp3"

print("Loading Whisper model: "+modelSize+" ...")
model = whisper.load_model(modelSize)

print("Transcribing: "+mp3+" ...")
result = model.transcribe(mp3)

print(result["text"])

print("Saving: "+mp3+".srt ...")
p = Path(mp3)
writer = WriteSRT(p.parent)
writer(result, mp3)
```

**Reply by archz2** [reply]
Okay. I used this code 

```
import sys
from pathlib import Path
import whisper
from whisper.utils import WriteSRT

modelSize = "large"

mp3 = "data/Scene1.mp3"

print("Loading Whisper model: "+modelSize+" ...")
model = whisper.load_model(modelSize)

print("Transcribing: "+mp3+" ...")
result = model.transcribe(mp3)

print(result["text"])

print("Saving: "+mp3+".srt ...")
p = Path(mp3)
writer = WriteSRT(p.parent)
writer(result, mp3)
```








and got the following error. 

![image](https://user-images.githubusercontent.com/66735186/218316183-48eb75d9-1201-408b-866c-5417f3d69c7d.png)

**Comment by EtienneAb3d** [comment]
"No such file or directory"
Please, try to investigate by yourself before asking for our time for such basic coding problem: the path you provided is not the good one. As I see on your left panel, your file is not in the sub-directory "data/".

**Reply by archz2** [reply]
When the Korean subtitles got generated my file was in 'sample_data' folder. I did not create any data folder. How do I ensure that the file inside 'sample_data' folder is used?

I tried an alternative. My file titled '02_02_Workflow.mp3' is in a folder called WhisperAudio in my Google drive. If I use the following address

"content/drive/MyDrive/WhisperAudio/02_02_Workflow.mp3"

I get the same error.
